crates/project/src/image_store.rs

```diff
@@ -52,9 +53,64 @@ pub struct ImageItem {
     pub file: Arc<dyn File>,
     pub image: Arc<gpui::Image>,
     reload_task: Option<Task<()>>,
+    pub width: Option<u32>,
+    pub height: Option<u32>,
+    pub file_size: Option<u64>,
+    pub color_type: Option<&'static str>,
+}
+
+fn image_color_type_description(color_type: ColorType) -> &'static str {
+    match color_type {
+        ColorType::L8 => "Grayscale (8-bit)",
+        ColorType::La8 => "Grayscale with Alpha (8-bit)",
+        ColorType::Rgba8 => "PNG (32-bit color)",
+        ColorType::Rgb8 => "RGB (24-bit color)",
+        ColorType::Rgb16 => "RGB (48-bit color)",
+        ColorType::Rgba16 => "PNG (64-bit color)",
+        ColorType::L16 => "Grayscale (16-bit)",
+        ColorType::La16 => "Grayscale with Alpha (16-bit)",
+
+        _ => "unknown color type",
+    }
 }
 
 impl ImageItem {
+    async fn image_info(
+        image: &ImageItem,
+        project: &Project,
+        cx: &AppContext,
+        fs: Arc<dyn fs::Fs>,
+    ) -> Result<(u32, u32, u64, &'static str), String> {
+        let worktree = project
+            .worktree_for_id(image.project_path(cx).worktree_id, cx)
+            .ok_or_else(|| "Could not find worktree for image".to_string())?;
+        let worktree_root = worktree.read(cx).abs_path();
+
+        let path = if image.path().is_absolute() {
+            image.path().to_path_buf()
+        } else {
+            worktree_root.join(image.path())
+        };
+
+        if !path.exists() {
+            return Err(format!("File does not exist at path: {:?}", path));
+        }
+
+        let img = image::open(&path).map_err(|e| format!("Failed to open image: {}", e))?;
```

Comment:
This should be wrapped in a `smol::unblock`, like here: https://github.com/zed-industries/zed/blob/2ff06cea6ee93b25e6d1d32ea247f70769207d82/crates/fs/src/fs.rs#L508-L509

Reply:
this too